guides/source/active_record_querying.md

```diff
@@ -592,6 +592,23 @@ Book.where("created_at >= :start_date AND created_at <= :end_date",
 
 This makes for clearer readability if you have a large number of variable conditions.
 
+#### Conditions That Use `LIKE`
+
+You can specify conditions that use SQL `LIKE` as following below.
+
+```ruby
+Book.where("title LIKE ?", "#{params[:title]}%")
+```
+
+Note that the example may not be enough because SQL `LIKE` characters (i.e., `%` and `_`) in `params[:title]` aren't sanitized. In this case, you can use `sanitize_sql_like` to sanitize SQL `LIKE` characters.
+
+```ruby
+Book.where("title LIKE ?",
+  "#{Book.sanitize_sql_like(params[:title])}%")
+```
+
+In this example, `params[:title]` is sanitized and `%` isn't sanitized. So, the query returns records which a title starts with `params[:title]` exactly even it includes any SQL `LIKE` characters.
```

Comment:
I was thinking we should start the section with the warning, so that the section's purpose is immediately obvious.

I took a stab at it to show what I mean:

``````suggestion
Although condition arguments are automatically escaped to prevent SQL injection, SQL `LIKE` wildcards (i.e., `%` and `_`) are **not** escaped. This may cause unexpected behavior if an unsanitized value is used in an argument. For example:

```ruby
Book.where("title LIKE ?", params[:title] + "%")
```

In the above code, the intent is to match titles that start with a user-specified string. However, any occurrences of `%` or `_` in `params[:title]` will be treated as wildcards, leading to surprising query results. In some circumstances, this may also prevent the database from using an intended index, leading to a much slower query.

To avoid these problems, use [`sanitize_sql_like`][] to escape wildcard characters in the relevant portion of the argument:

```ruby
Book.where("title LIKE ?", 
  Book.sanitize_sql_like(params[:title]) + "%")
```

[`sanitize_sql_like`]: https://api.rubyonrails.org/classes/ActiveRecord/Sanitization/ClassMethods.html#method-i-sanitize_sql_like
``````

Feel free to edit it as you like.  Also, though I would use string interpolation in real-world code, I decided to use `+ "%"` in the examples to make it easier to see the difference, but I'm open to changing it back.

By the way, whenever you're happy with all of the changes, would you mind [squashing the commits](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits)?

Reply:
Oh, I am sorry. Thank you very much. I rewrited it.